src/couch_index/src/couch_index.erl

```diff
@@ -192,7 +192,17 @@ handle_call({compacted, NewIdxState}, _From, State) ->
             end;
         false ->
             {reply, ok, commit_compacted(NewIdxState, State)}
-    end.
+    end;
+handle_call({compaction_failed, Reason}, _From, State) ->
+    #st{
+        mod = Mod,
+        idx_state = OldIdxState,
+        waiters = Waiters
+    } = State,
+    send_all(Waiters, Reason),
+    {ok, NewIdxState} = Mod:remove_compacted(OldIdxState),
```

Comment:
This is a braking change. It introduces a new API call which could be missing on custom index type. 
If the Mod doesn't have remove_compacted function it would crash here.
I think it is acceptable behavior though since it is currently failing anyway.